src/Http/Routing/src/Matching/HostMatcherPolicy.cs

```diff
@@ -0,0 +1,330 @@
+        {
+            if (endpoints == null)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            return endpoints.Any(e =>
+            {
+                var hosts = e.Metadata.GetMetadata<IHostMetadata>()?.Hosts;
+                if (hosts == null || hosts.Count == 0)
+                {
+                    return false;
+                }
+
+                foreach (var host in hosts)
+                {
+                    // Don't run policy on endpoints that match everything
+                    var key = CreateEdgeKey(host);
+                    if (!key.MatchesAll)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            });
+        }
+
+        private static EdgeKey CreateEdgeKey(string host)
+        {
+            if (host == null)
+            {
+                return EdgeKey.WildcardEdgeKey;
+            }
+
+            var hostParts = host.Split(':');
+            if (hostParts.Length == 1)
+            {
+                return new EdgeKey(hostParts[0], null);
+            }
+            if (hostParts.Length == 2)
+            {
+                if (int.TryParse(hostParts[1], out var port))
+                {
+                    return new EdgeKey(hostParts[0], port);
+                }
+                else if (string.Equals(hostParts[1], "*", StringComparison.Ordinal))
+                {
+                    return new EdgeKey(hostParts[0], null);
+                }
+            }
+
+            throw new InvalidOperationException($"Could not parse host: {host}");
+        }
+
+        public IReadOnlyList<PolicyNodeEdge> GetEdges(IReadOnlyList<Endpoint> endpoints)
+        {
+            if (endpoints == null)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            // The algorithm here is designed to be preserve the order of the endpoints
+            // while also being relatively simple. Preserving order is important.
+
+            // First, build a dictionary of all of the hosts that are included
+            // at this node.
+            //
+            // For now we're just building up the set of keys. We don't add any endpoints
+            // to lists now because we don't want ordering problems.
+            var edges = new Dictionary<EdgeKey, List<Endpoint>>();
+            for (var i = 0; i < endpoints.Count; i++)
+            {
+                var endpoint = endpoints[i];
+                var hosts = endpoint.Metadata.GetMetadata<IHostMetadata>()?.Hosts.Select(h => CreateEdgeKey(h)).ToArray();
+                if (hosts == null || hosts.Length == 0)
+                {
+                    hosts = new[] { EdgeKey.WildcardEdgeKey };
+                }
+
+                for (var j = 0; j < hosts.Length; j++)
+                {
+                    var contentType = hosts[j];
```

Comment:
contentType?

Reply:
Leftover copypasta from the ConsumesPolicyMatcher